### ISSUE TITLE
[prisma] Support for Prisma V5

### ISSUE THREAD

**Issue by adri1wald** [stalniy/casl#788]
**Describe the bug**
Prisma v5 is not supported.

**To Reproduce**
1. Try to upgrade to prisma and @prisma/client 5.0.0

**Expected behavior**
Be able to upgrade to prisma v5

**CASL Version**
```json
{
    "@casl/ability": "^6.3.3",
    "@casl/prisma": "^1.3.3"
}
```

**Comment by kriss-u** [comment]
I forcefully upgraded the prisma packages and it didn't give me any error. Right now, I am using conditions and field matcher in my project.

Can anyone else confirm any regressions due to package upgrade? If that's the case, we might be able to add prisma@5 to the dependency list.

**Comment by huthaifaa** [comment]
Same here, updating the peerDeps to `|| ^5.0.0` didn't break functionality

**Comment by huthaifaa** [comment]
> I forcefully upgraded the prisma packages and it didn't give me any error. Right now, I am using conditions and field matcher in my project.
> 
> Can anyone else confirm any regressions due to package upgrade? If that's the case, we might be able to add prisma@5 to the dependency list.

Same here, updating the peerDeps to || ^5.0.0 didn't break functionality. Hopefully it's a quick pull request

**Comment by msmps** [comment]
I bumped my prisma version to v5 and it looks to have broken some of the typescript types.

```bash
@x/permissions:type-check:     Argument of type '{ id: number; }' is not assignable to parameter of type 'string | string[] | undefined'.
@x/permissions:type-check:       Object literal may only specify known properties, and 'id' does not exist in type 'string[]'.
```

![image](https://github.com/stalniy/casl/assets/7691252/36fe34bf-85de-4d36-87c1-bb6c3626fe86)

**Comment by jd4u** [comment]
```
"@casl/ability": "^6.5.0"
"@casl/prisma": "^1.4.0",
```

Using --force, the upgrade to Prisma 5 is smooth for me. No issues in code anywhere.

**Comment by adri1wald** [comment]
Only change I can see that could potentially interact with casl is https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-5#removal-of-array-shortcuts. But it seems the type is only narrowed not expanded so maybe internally within casl there are no problems but consumers may need to adapt their code. So it would constitute a breaking change like Prisma 5 is itself.

**Comment by jd4u** [comment]
> ```
> "@casl/ability": "^6.5.0"
> "@casl/prisma": "^1.4.0",
> ```
> 
> Using --force, the upgrade to Prisma 5 is smooth for me. No issues in code anywhere.

Conflicts happens while installing any new package or upgrading because of version dependencies..

Below is the output after using --force with npm install/upgrade commands...

![image](https://github.com/stalniy/casl/assets/1511611/7d55427d-9bbf-47d7-b741-d985c580de43)

**Comment by jd4u** [comment]
@stalniy , it is appreciated if the Prisma 5 support is released soon.

**Comment by stalniy** [comment]
released in @casl/prisma@1.4.1

**Comment by stalniy** [comment]
please create another issue if you find any inconsistencies
